### ISSUE TITLE
Implement model updates subgraph on device

### ISSUE THREAD

**Issue by boris-drazic** [tenstorrent-metal/tt-metal#5717]
Create a list of all OPs that are executing on host between iterations of UNet model.
We would like all OPs that are in the loop to execute on device, so disregard OPs that don't need to be in the loop (i.e., pre-processing that can be done outside of the loop or saving images).

**Comment by Sudharsan-V** [comment]
@boris-drazic, The following ops are part of the scheduler that are used in loop.

- torch.expand
- torch.chunk

**Comment by boris-drazic** [comment]
We can use split for chunk and I don't think expand is needed if we precompute all that we can.

`ttnn_scheduler.timesteps` is known at the start of the loop, thus the following three lines can be precomputed outside the loop
```
  _t = constant_prop_time_embeddings(t, ttnn_latent_model_input, unet.time_proj)
  _t = _t.unsqueeze(0).unsqueeze(0)
  _t = ttnn.from_torch(_t, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT, device=device)
```

`preprocess_model_parameters` should also move outside of the loop since it is pre-processing

`noise_pred = guide(noise_pred, guidance_scale, t)` use ttnn OPs to implement this

`ttnn_latents = ttnn_scheduler.step(noise_pred, t, ttnn_latents).prev_sample` this step of scheduler should also be implemented with ttnn on device. Looking inside of it, looks like `lms_coeffs` can be precomputed as it depends on `step_index` and `order`. And then the loop `coeff * derivative for coeff, derivative in zip(lms_coeffs, reversed(self.derivatives))` should be replaced with tensor math: multiply and add `lms_coeffs` and `self.derivatives`.

With this we should also get rid of moving tensors to/from PyTorch.

**Comment by vigneshkeerthivasanx** [comment]
The PR #5477 has been updated with the test_performance file which currently uses `models/experimental/functional_stable_diffusion/tt2.` along with all the scheduler functions being used in ttnn

**Comment by Sudharsan-V** [comment]
The suggested inputs have been included, and the scheduler is now enabled on the device. The changes have been added to the PR https://github.com/tenstorrent-metal/tt-metal/pull/5477.

**Comment by Sudharsan-V** [comment]
Completed implementing the scheduler class using ttnn ops and it is merged with the main. This is included in the PR [#5699](https://github.com/tenstorrent-metal/tt-metal/pull/5699)
Hence closing this ticket.
